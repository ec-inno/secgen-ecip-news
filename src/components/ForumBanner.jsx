import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import BackgroundImage from 'gatsby-background-image';

import Button from '../components/Button';

const ForumBanner = () => {
  const icon = {
    shape: 'ui--corner-arrow',
    size: 'xs',
    transform: 'rotate-90',
  };

  const data = useStaticQuery(graphql`
    query {
      desktop: file(relativePath: { eq: "bg-forum.png" }) {
        childImageSharp {
          fluid(quality: 90, maxWidth: 4160) {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
    }
  `);

  return (
    <>
      <BackgroundImage
        Tag="section"
        className="ecl-page-banner ecl-page-banner--image-shade ecl-page-banner--centered ecl-u-mt-l"
        fluid={data.desktop.childImageSharp.fluid}
        backgroundColor="#000"
      >
        <div className="ecl-container ecl-page-banner__container">
          <div className="ecl-page-banner__content">
            <h1 className="ecl-page-banner__title">
              Want to learn and collaborate?
            </h1>
            <Button
              className="ecl-page-banner__button"
              variant="call"
              label="Join the forum"
              icon={icon}
              iconPosition="after"
            />
          </div>
        </div>
      </BackgroundImage>
    </>
  );
};

export default ForumBanner;
