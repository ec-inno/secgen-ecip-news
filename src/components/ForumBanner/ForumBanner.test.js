import React from 'react';
import renderer from 'react-test-renderer';
import ForumBanner from './ForumBanner';

describe('ForumBanner', () => {
  const props = {
    label: 'test',
  };

  it('renders correctly', () => {
    const tree = renderer.create(<ForumBanner {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
