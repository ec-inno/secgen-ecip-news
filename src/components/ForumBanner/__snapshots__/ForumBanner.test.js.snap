// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForumBanner renders correctly 1`] = `
<section
  className="ecl-page-banner ecl-page-banner--image-shade ecl-page-banner--centered ecl-u-mt-l"
>
  <div
    className="ecl-page-banner__image"
    style={
      Object {
        "backgroundImage": "url('file')",
      }
    }
  />
  <div
    className="ecl-container ecl-page-banner__container"
  >
    <div
      className="ecl-page-banner__content"
    >
      <h1
        className="ecl-page-banner__title"
      >
        Want to learn and collaborate?
      </h1>
      <a
        className="ecl-page-banner__button ecl-link ecl-link--cta ecl-link--icon ecl-link--icon-after"
        href="#"
      >
        <span
          className="ecl-link__label"
        >
          Join the forum
        </span>
        Â 
        <svg
          aria-hidden="true"
          className="ecl-link__icon ecl-icon ecl-icon--xs ecl-icon--rotate-90"
          focusable="false"
        >
          
          
          <use
            xlinkHref="file#ui--corner-arrow"
          />
        </svg>
      </a>
    </div>
  </div>
</section>
`;
